workshops/views.py

```diff
@@ -178,6 +178,16 @@ class SiteUpdate(LoginRequiredMixin, UpdateViewContext):
     slug_url_kwarg = 'site_domain'
     template_name = 'workshops/generic_form.html'
 
+
+@login_required
+def site_delete(request, site_domain):
+    """Delete specific site."""
+    site = get_object_or_404(Site, domain=site_domain)
+    site.delete()
```

Comment:
I tend to prefer idempotent deletes, because it seems odd to have one of a pair of racy deletes return a 404 because the object has already been deleted.  The deleter just wants to remove the item from play, and if it's already gone (even if it never existed) that seems like success to me ;).  In Django, that's:

    Site.objects.filter(domain=site_domain).delete()